lib/ast/AST.cpp

```diff
@@ -1,11 +1,33 @@
 #include "kllvm/ast/AST.h"
 
+#include <unordered_set>
+
 using namespace kllvm;
 
+bool KOREObjectSortVariable::operator==(const KOREObjectSort &other) const {
+  if (auto var = dynamic_cast<const KOREObjectSortVariable *>(&other)) {
+    return var->name == name;
+  }
+  return false;
+}
+
 void KOREObjectCompositeSort::addArgument(KOREObjectSort *Argument) {
   arguments.push_back(Argument);
 }
 
+bool KOREObjectCompositeSort::operator==(const KOREObjectSort &other) const {
+  if (auto sort = dynamic_cast<const KOREObjectCompositeSort *>(&other)) {
+    if (sort->name != name || sort->arguments.size() != arguments.size()) {
+      return false;
+    }
+    for (int i = 0; i < arguments.size(); i++) {
```

Comment:
super nitpicky, but prefer ++i to i++.

Reply:
yes I tend to agree. Typically it only matters when ++ is overloaded for some iterator, where the prefix does an in place update and hence is a bit more efficient, but it's nice to be consistent.